web/src/transmit.js

```diff
@@ -70,8 +71,18 @@ export function send_message(request, on_success, error) {
                     return;
                 }
 
-                const response = channel.xhr_error_message("Error sending message", xhr);
-                error(response);
+                let response;
+                let html_content = false;
+                if (xhr.responseJSON?.code === "TOPIC_WILDCARD_MENTION_NOT_ALLOWED") {
+                    response = $t_html({
+                        defaultMessage:
+                            "You do not have permission to use <b>@topic</b> mentions in this topic.",
+                    });
+                    html_content = true;
+                } else {
+                    response = channel.xhr_error_message("Error sending message", xhr);
+                }
+                error(response, html_content);
```

Comment:
This may be a bit more awkward to plumb to the `compose_banner` API. I think one option is to pass to `error` the `code` parameter, and call a new wrapper for `compose_banner.show_error_message` called `compose_banner.show_server_send_failure` which has an extra `server_error_code` parameter, and renders the topic mentions banner if that's appropriate? Not sure, maybe there's a simpler way to do it, but this should work, and I think would not be a bad API change to have a dedicated function for processing server errors.

Reply:
Followed the[ same approach](https://github.com/zulip/zulip/pull/27813#discussion_r1402739099) as discussed for message edit.